src/templates/dandelion/config/screens/LockScreen.js

```diff
@@ -0,0 +1,338 @@
+          <TokenSelector
+            selectedIndex={lockToken.selectedIndex}
+            onChange={handleTokenChange}
+            value={lockToken.data.address}
+            tokens={[DEFAULT_LOCK_TOKEN]}
+          />
+        </Field>
+
+        <Field
+          label={
+            <React.Fragment>
+              Lock amount
+              <Help hint="What’s the lock amount?">
+                <strong>Lock amount</strong> is the amount of tokens to lock.
+              </Help>
+            </React.Fragment>
+          }
+        >
+          <TextInput
+            onChange={handleLockAmountChange}
+            value={lockAmount === -1 ? '' : lockAmount}
+            wide
+          />
+        </Field>
+      </div>
+
+      <Duration
+        duration={lockDuration}
+        onUpdate={handleDurationChange}
+        label={
+          <React.Fragment>
+            {DURATION_LABEL}
+            <Help hint="What’s the lock duration?">{DURATION_HELP_TEXT}</Help>
+          </React.Fragment>
+        }
+      />
+
+      <Field
+        label={
+          <React.Fragment>
+            Spam penalty %
+            <Help hint="What’s the spam penalty?">
+              <strong>Spam penalty</strong>
+              Is a % of the base lock amount and lock duration. The more active
+              locks an account has, the more tokens will have to lock and for a
+              longer period of time for subsequent proposals.
+            </Help>
+          </React.Fragment>
+        }
+      >
+        <TextInput
+          type="number"
+          step={1}
+          onChange={handleSpamPenaltyChange}
+          value={spamPenalty}
+        />
+        <span> %</span>
+      </Field>
+
+      {formError && (
+        <Info
+          mode="error"
+          css={`
+            margin-bottom: ${3 * GU}px;
+          `}
+        >
+          {formError}
+        </Info>
+      )}
+
+      <Info
+        css={`
+          margin-bottom: ${3 * GU}px;
+        `}
+      >
+        These settings will determine the token to be locked, how long tokens
+        are locked, the amount of tokens to lock and a penalty percentage for
+        spamming proposals. At initialization the token address parameter can be
+        set to an ERC20 token. It cannot be changed. If a change is necessary
+        the user can install a new instance and change permissions in the
+        organization to reflect the change.
+      </Info>
+
+      <Navigation
+        ref={prevNextRef}
+        backEnabled
+        nextEnabled
+        nextLabel={`Next: ${screens[screenIndex + 1][0]}`}
+        onBack={back}
+        onNext={handleSubmit}
+      />
+    </form>
+  )
+}
+
+LockScreen.propTypes = {
+  appLabel: PropTypes.string,
+  dataKey: PropTypes.string,
+  screenProps: ScreenPropsType.isRequired,
+  title: PropTypes.string,
+}
+
+LockScreen.defaultProps = {
+  appLabel: 'Time Lock',
+  dataKey: 'lock',
+  title: 'Configure template',
+}
+
+function formatDuration(duration) {
```

Comment:
It looks like we're starting to use this a lot, could we move this into a `kit-utils.js`?